src/pcre2_util.h

```diff
@@ -44,26 +44,52 @@ POSSIBILITY OF SUCH DAMAGE.
 /* Assertion macros */
 
 #ifdef HAVE_BUILTIN_UNREACHABLE
-#define PCRE2_UNREACHABLE() __builtin_unreachable()
-#elif HAVE_BUILTIN_ASSUME
-#define PCRE2_UNREACHABLE() __assume(0)
+#define PCRE2_DEADTRAP() __builtin_unreachable()
+#elif defined(HAVE_BUILTIN_ASSUME)
+#define PCRE2_DEADTRAP() __assume(0)
 #else
-#define PCRE2_UNREACHABLE() do {} while(0)
+#warning "missing implementation of PCRE2_UNREACHABLE()"
 #endif
 
 #ifdef PCRE2_DEBUG
+
 #if defined(HAVE_ASSERT_H) && !defined(NDEBUG)
 #include <assert.h>
 #define PCRE2_ASSERT(x) assert(x)
 #elif defined(HAVE_STDLIB_H) && defined(HAVE_STDIO_H)
```

Comment:
this additional checks were apparently introduced by mistake, which is why the final version might also remove them and why they are not being used in the new debug assert for PCRE2_UNREACHABLE() below.

the core was just reformatted so it can be easier to read and also compare with the proposed similar implementation.